### ISSUE TITLE
CLI doesn't render double-wide characters properly

### ISSUE THREAD

**Issue by martint** [facebook/presto#3333]
```
presto> select * from (VALUES ('网',1),('x',2));
 _col0 | _col1 
-------+-------
 网     |     1 
 x     |     2 
(2 rows)
```

**Comment by yuananf** [comment]
I'm trying to fix it, and I want to use ```Slices.utf8Slice``` to solve it.
But I face a problem, '网' has 3 bytes when encoded to UTF8. But it was rendered to 2 bytes.
Any other way to get the real byte length of the String.

**Comment by electrum** [comment]
I would love if someone could find some documentation about how terminals work regarding wide characters.

**Comment by electrum** [comment]
https://en.m.wikipedia.org/wiki/Halfwidth_and_fullwidth_forms

**Comment by electrum** [comment]
Recently added to JLine: https://github.com/jline/jline2/blob/master/src/main/java/jline/console/WCWidth.java

**Comment by electrum** [comment]
This is also broken for control characters:
```
presto:default> select chr(0), 'x';
 _col0 | _col1 
-------+-------
 ^@     | x     
(1 row)

**Comment by electrum** [comment]
Fixed in #3596 (will open a separate issue for control characters)
